src/openzeppelin/token/erc721_enumerable/library.cairo

```diff
@@ -196,23 +196,22 @@ func _remove_token_from_all_tokens_enumeration{
     let (supply: Uint256) = ERC721_Enumerable_all_tokens_len.read()
     let (last_token_index: Uint256) = SafeUint256.sub_le(supply, Uint256(1, 0))
     let (token_index: Uint256) = ERC721_Enumerable_all_tokens_index.read(token_id)
-
-    # When the token to delete is the last token, the swap operation is unnecessary. However,
-    # since this occurs so rarely (when the last minted token is burnt), we still do the swap
-    # here to avoid the gas cost of adding an 'if' statement (like in _remove_token_from_owner_enumeration)
     let (last_token_id: Uint256) = ERC721_Enumerable_all_tokens.read(last_token_index)
 
     ERC721_Enumerable_all_tokens.write(last_token_index, Uint256(0, 0))
-    ERC721_Enumerable_all_tokens.write(token_index, last_token_id)
-
-    ERC721_Enumerable_all_tokens_index.write(last_token_id, token_index)
     ERC721_Enumerable_all_tokens_index.write(token_id, Uint256(0, 0))
+    ERC721_Enumerable_all_tokens_len.write(last_token_index)
 
-    let (new_supply: Uint256) = SafeUint256.sub_le(supply, Uint256(1, 0))
```

Comment:
heh benchmarking might be tricky: how do we know adding the `if` isn't less performant, but we don't notice due this line making up for it?